gitcoinco/grants-stack
# Fund Contract for Allo V1 and V2

fixes #2901 
fixes https://github.com/gitcoinco/grants-stack/issues/2900

# Description 

This implements funding the round for both V1 and V2.

I have also made some copy changes, because in Allo V2, you do not really fund a "round contract" and this is not something you can do by just transferring tokens to the "round contract". Allo V2 has a single contract for all rounds and you fund them by calling a shared contract method. The changes are:

- Rename of the "Fund contract" section to "Fund round"
- Tweak paragraph for Allo v2 that mentions that you can transfer tokens to the round address

## How to test

For both Allo v1 and v2:

- Open the "Fund Contract" section in Manager
- Matching pool size should be correct
- Should not allow you to fund an amount that you do not have
- Try to fund the contract (test both a round with a native token and a ERC20)
- Funded amount should increase accordingly

I made some rounds called "QF Round ERC20" partially funded that should be useful to test this

## Checklist

This PR:

- [ ] Does it add new payout or donation token? In this case, have they been added to the indexer to avoid it to crash?
